### PR TITLE
Provide players adequate thinking time and warnings

### PR DESCRIPTION
**Pull request by ddugovic**
Eventually I partially figured out how to test this, and there are only two issues that I've observed:
* The "cache" collection contains references to superblitz (now hyperblitz)
* The tournament schedule may or may not present any problems due to existing tournaments (I suspect not because tournaments link to category IDs, right?)

```
db.cache.find()
{ "_id" : "user:top:week:", "v" : [ ], "e" : ISODate("2020-11-18T00:35:48.963Z") }
{ "_id" : "tournament:winner:all:", "v" : { "hyperbullet" : {  }, "bullet" : {  }, "superblitz" : {  }, "blitz" : {  }, "rapid" : {  }, "elite" : [ ], "marathon" : [ ], "variants" : { "chess960" : {  }, "antichess" : {  }, "horde" : {  }, "atomic" : {  }, "racingKings" : {  }, "kingOfTheHill" : {  }, "crazyhouse" : {  }, "threeCheck" : {  } } }, "e" : ISODate("2020-11-18T01:25:48.998Z") }
{ "_id" : "user:top:nbGame:", "v" : [ { "user" : { "id" : "toadofsky", "name" : "Toadofsky", "isPatron" : false }, "count" : 0 } ], "e" : ISODate("2020-11-24T11:11:04.943Z") }
{ "_id" : "game:imported:toadofsky", "v" : 0, "e" : ISODate("2020-12-24T10:40:25.719Z") }
{ "_id" : "history:rating:toadofsky", "v" : "", "e" : ISODate("2020-11-24T11:40:25.756Z") }
```

### CODE REVIEW

**Comment by ddugovic** [comment]
@WandererXII Please review.  Note this change does remove the HyperBullet shield because we do not want players to become addicted to playihg games which last less than 1 minute.

File: app/templating/TournamentHelper.scala [hunk 1]
```diff
@@ -50,7 +50,8 @@ trait TournamentHelper { self: I18nHelper with DateHelper with UserHelper =>
       "Lishogi "    -> "",
       "Marathon"    -> icon('\\'),
       "HyperBullet" -> s"H${icon(PerfType.Bullet.iconChar)}",
-      "SuperBlitz"  -> s"S${icon(PerfType.Blitz.iconChar)}"
+      "HyperBlitz"  -> s"S${icon(PerfType.Blitz.iconChar)}",
```
Comment (ddugovic) [inline app/templating/TournamentHelper.scala:53]: Hm, perhaps this is problematic due to existing data somewhere:

```
[HandlerException: Fails to handle v: Fails to handle hyperblitz: The key 'hyperblitz' could not be found in document {
  'hyperbullet': {

  },
  'bullet': {

  },
  'superblitz': {

  },
...
```

**Comment by justremember** [comment]
Oh, can you at least describe in detail what db changes wanderer needs to do in production if this gets pushed?

**Comment by ddugovic** [comment]
Yeah, I'm not fluent in mongoDB but I'll figure out how to update whatever data needs to be updated... I might as well livestream my attempt to figure that out.

**Comment by ddugovic** [comment]
OK, I found some entries (aside from the obvious: somehow `tournament2` would also need to be updated if we want to rename "superblitz" to "hyperblitz" so we can have "hyperbullet" and "hyperblitz" and "hyperrapid"):

```
> db.activity.find()
{ "_id" : "toadofsky:84", "l" : { "intoduction" : 2 }, "e" : [ "shogi-students" ] }
{ "_id" : "toadofsky:86", "l" : { "intoduction" : 2 } }
> db.cache.find()
{ "_id" : "user:top:week:", "v" : [ ], "e" : ISODate("2020-11-18T00:35:48.963Z") }
{ "_id" : "tournament:winner:all:", "v" : { "hyperbullet" : {  }, "bullet" : {  }, "superblitz" : {  }, "blitz" : {  }, "rapid" : {  }, "elite" : [ ], "marathon" : [ ], "variants" : { "chess960" : {  }, "antichess" : {  }, "horde" : {  }, "atomic" : {  }, "racingKings" : {  }, "kingOfTheHill" : {  }, "crazyhouse" : {  }, "threeCheck" : {  } } }, "e" : ISODate("2020-11-18T01:25:48.998Z") }
{ "_id" : "user:top:nbGame:", "v" : [ { "user" : { "id" : "toadofsky", "name" : "Toadofsky", "isPatron" : false }, "count" : 0 } ], "e" : ISODate("2020-11-24T11:11:04.943Z") }
{ "_id" : "game:imported:toadofsky", "v" : 0, "e" : ISODate("2020-12-24T10:40:25.719Z") }
{ "_id" : "history:rating:toadofsky", "v" : "", "e" : ISODate("2020-11-24T11:40:25.756Z") }
```

My quick fix here was:

```
db.cache.remove({})
WriteResult({ "nRemoved" : 5 })
```

File: modules/chess/src/main/scala/Clock.scala [hunk 2]
```diff
@@ -254,10 +255,9 @@ object Clock {
     def limitString: String =
       limitSeconds match {
         case l if l % 60 == 0 => (l / 60).toString
-        case 15 => "¼"
         case 30 => "½"
-        case 45 => "¾"
```
Comment (WandererXII) [inline modules/chess/src/main/scala/Clock.scala:259]: I can see, why you would want to get rid of these times for byo, but they might still be useful for inc. And then there is this whole thing of backward compatibility.

File: modules/game/src/main/BinaryFormat.scala [hunk 3]
```diff
@@ -170,14 +170,14 @@ object BinaryFormat {
       // The database expects a byte for a limit, and this is limit / 60.
       // For 0.5+0, this does not give a round number, so there needs to be
       // an alternative way to describe 0.5.
-      // The max limit where limit % 60 == 0, returns 180 for limit / 60
+      // The max classical limit where limit % 60 == 0, returns 180 (minutes).
       // So, for the limits where limit % 30 == 0, we can use the space
-      // from 181-255, where 181 represents 0.25 and 182 represents 0.50...
-      (if (limit % 60 == 0) limit / 60 else limit / 15 + 180).toByte
+      // from 181-255, where 181 represents 0.5 and 183 represents 1.5.
+      (if (limit % 60 == 0) limit / 60 else limit / 30 + 180).toByte
```
Comment (WandererXII) [inline modules/game/src/main/BinaryFormat.scala:176]: I wouldn't really change this, relates to my previous comment.

**Comment by WandererXII** [comment]
Sorry, if it seemed like I was ignoring this PR, that definitely wasn't my intention, there are some really nice and needed changes here. It just seemed like it required some testing and db fixing and making sure that there are no further conflicts with existing data on my part and I didn't have enough time for that. Sorry about that, I made some comments, all looks good, I would just prefer to keep the current init limits (15s, 45s), if that's fine with you I will merge this.

File: modules/chess/src/main/scala/Clock.scala (hunk 2 above)
Comment (ddugovic) [reply modules/chess/src/main/scala/Clock.scala:259]: Ah, right, increment... OK, yes between that and backward compatibility I'll revert the 1/4 and 3/4 change.

**Comment by ddugovic** [comment]
Many thanks for your consideration; I understand software is complicated and am very grateful for your help and consideration.  Thanks!!

**Comment by ddugovic** [comment]
It appears I'll need to somehow fix the UI too:

**Comment by justremember** [comment]
What's the mongodb command to fix this error? @WandererXII @ddugovic
First time doing `./lila run` since this was pushed.
```
[HandlerException: Fails to handle 'v': Fails to handle 'hyperrapid': The key 'hyperrapid' could not be found in document {
  'hyperbullet': {

  },
  'bullet': {

  },
  'superblitz': {

  },
  'blitz': {

  },
  'rapid': {

  },
  'elite': [

  ],
  'marathon': [

  ],```

**Pull request merged**